### ISSUE TITLE
Feature request: Ability to monitor expired domains using negative $warning and $critical values

### ISSUE THREAD

**Issue by Elyrith** [glensc/nagios-plugin-check_domain#32]
One of my friends had their domain registered by somebody else long ago and now it's expired and they can't get control of it to renew. The domain has status "redemptionPeriod http://www.icann.org/epp#redemptionPeriod". They want me to wait until it's deleted and snatch it up so they continue using it. I looked up the registrar and I know the values for approximately when it's going to be released/deleted.

I have two options:
1. I can create a little plugin to monitor the "status" of the domain and notify me when it enters or exits a specific status, or...
2. I could use this plugin but use negative values for $warning and $critical

I've created a branch with the modified logic. I'll create a pull request so you can take a look.

Also, let me know if you think that this functionality doesn't really fit with what this plugin is meant to do. I'm not entirely sure myself. :smile_cat:

**Comment by glensc** [comment]
you can use `negate` plugin from `nagios-plugins` distribution to invert:

```
$ /usr/lib/nagios/plugins/check_dummy 0 'oh long johnson'; echo $?
OK: oh long johnson
0
$ /usr/lib/nagios/plugins/negate /usr/lib/nagios/plugins/check_dummy 0 'oh long johnson'; echo $?
OK: oh long johnson
2
$ /usr/lib/nagios/plugins/negate /usr/lib/nagios/plugins/check_dummy 2 'oh long johnson'; echo $?
CRITICAL: oh long johnson
0

```

**Comment by Elyrith** [comment]
Because check_domain's first logic check is "if it's expired, exit with CRITICAL and say it expired XX days ago", that means if I use negate then the logic will change to "if it's expired, exit with OK and say it expired XX days ago".

What I want is an OK result if it expired <70 days ago (because the registrar has 40 days in 'expired step 1' state and 30 days in 'expired step 2'), then only report CRITICAL if it expired <=70 days ago. This way I'll know when to buy up the officially deleted domain with another registrar.

Like I mentioned, this may be outside the scope of what this plugin should do, but in my mind it's related enough and all the code is the same.

**Comment by glensc** [comment]
i didn't get thinking too much of into it, so i gave you pointer to "negate" plugin. in fact, what it "says" and what it "returns" (the exit code) are different.

but if you still say, negate can't solve your issue (esthetically it may be wrong to say OK but return ERROR statys code), i'm ok with merging the PR.

**Comment by glensc** [comment]
merged via #33
